pkg/testing/testerrors/testerrors.go

```diff
@@ -29,12 +31,47 @@ import (
 // only compares the error code. Using testerrors.Equal, the assertion is more
 // strict.
 func AssertEqual(t *testing.T, expected, actual error, msgAndArgs ...interface{}) {
-	testutil.AssertEqual(t, testableError(expected), actual, msgAndArgs...)
+	testutil.AssertEqual(t, testableError(expected), testableError(actual), msgAndArgs...)
 }
 
 func testableError(err error) error {
 	if err == nil {
 		return nil
 	}
-	return testutil.EqualError(err)
+	// TODO: replace with testutil.EqualError after cli-utils is updated for asymmetric comparison
+	return EqualError(err)
+}
+
+// EqualError returns an error with an Is(error)bool function that matches
+// any error with the same type and string value as the supplied error.
+//
+// Use with AssertEqual to handle error comparisons.
+func EqualError(err error) error {
+	return &equalError{
+		err: err,
+	}
+}
+
+type equalError struct {
+	err error
+}
+
+func (e *equalError) Error() string {
+	return fmt.Sprintf("EqualError{Type: %T, Error: %q}", e.err, e.err)
```

Comment:
Type and error both e.err?

Reply:
%T prints the type of the error. %q prints the Error() string message.